dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/SourcesToStartingPoints.scala

```diff
@@ -9,48 +9,131 @@ import io.shiftleft.semanticcpg.language.operatorextension.allAssignmentTypes
 import io.shiftleft.semanticcpg.utils.MemberAccess.isFieldAccess
 import org.slf4j.LoggerFactory
 
-import java.util.concurrent.{ForkJoinPool, ForkJoinTask, RecursiveTask, RejectedExecutionException}
+import java.util.concurrent.*
+import scala.collection.mutable.ListBuffer
 import scala.util.{Failure, Success, Try}
 
 case class StartingPointWithSource(startingPoint: CfgNode, source: StoredNode)
-
+case class UsageInput(src: StoredNode, typeDecl: TypeDecl, astNode: AstNode)
+case class ResultSummary(result: List[StartingPointWithSource], methodTasks: List[UsageInput])
 object SourcesToStartingPoints {
 
   private val log = LoggerFactory.getLogger(SourcesToStartingPoints.getClass)
 
   def sourceTravsToStartingPoints[NodeType](sourceTravs: IterableOnce[NodeType]*): List[StartingPointWithSource] = {
-    val fjp = ForkJoinPool.commonPool()
+    val executorService = Executors.newWorkStealingPool()
+    val completionService =
```

Comment:
I wonder if the completion service is necessary, it's not being used any differently than submitting or invoking directly on the `executorService`

Reply:
that makes sense. I didn't question myself on that. Let me try removing it, as we are not waiting for the results to be fetched from the queue, unlike in `Engine`

Reply:
Made the change